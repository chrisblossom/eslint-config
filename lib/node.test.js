'use strict';

const path = require('path');
const readDirDeep = require('read-dir-deep');
const Eslint = require('eslint').CLIEngine;

const sandbox = path.resolve(__dirname, '__sandbox__/node/');

const files = readDirDeep
    .sync(sandbox)
    .map((file) => path.resolve(sandbox, file))
    .filter((file) => path.parse(file).ext === '.js');

let eslint;

const cwd = process.cwd();
beforeEach(() => {
    process.chdir(sandbox);

    eslint = new Eslint({
        baseConfig: require('./node'),
        useEslintrc: false,
        ignore: false,
    });
});

afterEach(() => {
    process.chdir(cwd);
});

describe('node', () => {
    test('lints javascript files', () => {
        const result = eslint.executeOnFiles(files).results;

        expect(result).toMatchSnapshot();
    });
});
