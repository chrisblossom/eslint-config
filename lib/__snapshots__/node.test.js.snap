// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node lints javascript files 1`] = `
Array [
  Object {
    "errorCount": 2,
    "filePath": "<PROJECT_ROOT>/lib/a.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 1,
        "endColumn": 20,
        "endLine": 6,
        "line": 1,
        "message": "Use the global form of 'use strict'.",
        "messageId": "global",
        "nodeType": "Program",
        "ruleId": "strict",
        "severity": 2,
      },
      Object {
        "column": 7,
        "endColumn": 13,
        "endLine": 1,
        "line": 1,
        "message": "'unused' is assigned a value but never used.",
        "nodeType": "Identifier",
        "ruleId": "no-unused-vars",
        "severity": 2,
      },
    ],
    "source": "const unused = 1;

function a() {
}

module.exports = a;
",
    "warningCount": 0,
  },
  Object {
    "errorCount": 2,
    "filePath": "<PROJECT_ROOT>/lib/b.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 1,
        "endColumn": 20,
        "endLine": 7,
        "line": 1,
        "message": "Use the global form of 'use strict'.",
        "messageId": "global",
        "nodeType": "Program",
        "ruleId": "strict",
        "severity": 2,
      },
      Object {
        "column": 1,
        "endColumn": 2,
        "endLine": 5,
        "line": 3,
        "message": "Async functions are not supported until Node.js 7.6.0. The configured version range is '>=6.9.0'.",
        "messageId": "no-async-functions",
        "nodeType": "FunctionDeclaration",
        "ruleId": "node/no-unsupported-features/es-syntax",
        "severity": 2,
      },
    ],
    "source": "const a = require('./a');

async function b() {
    a();
}

module.exports = b;
",
    "warningCount": 0,
  },
  Object {
    "errorCount": 1,
    "filePath": "<PROJECT_ROOT>/lib/c.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 1,
        "fatal": true,
        "line": 1,
        "message": "Parsing error: 'import' and 'export' may appear only with 'sourceType: module'",
        "ruleId": null,
        "severity": 2,
      },
    ],
    "source": "import a from './a';

async function b() {
    a();
}

export { b };
",
    "warningCount": 0,
  },
  Object {
    "errorCount": 0,
    "filePath": "<PROJECT_ROOT>/lib/node.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "warningCount": 0,
  },
]
`;
